sandbox/lua/encoders/schema_influx_write.lua

```diff
@@ -0,0 +1,319 @@
+    else
+        name_prefix = name_prefix_orig
+    end
+
+    -- Initialize and populate the table of tags to include in the
+    -- InfluxDB write API message
+    -- Convert value to a string as this is required by the API
+    local message_tags = {}
+    for _, field in ipairs(base_fields_list) do
+        if (used_tag_fields["**all_base**"] or used_tag_fields["**all_base**"])
+            and base_fields_tag_map[field] then
+            message_tags[field] = tostring(read_message(field))
+        elseif used_tag_fields[field] and base_fields_tag_map[field] then
+            message_tags[field] = tostring(read_message(field))
+        end
+    end
+
+    -- Initialize a counter for the InfluxDB write API message data points
+    local points_index = 1
+
+    -- Initialize the table of data points and populate it with data
+    -- from the Heka message.  When skip_fields includes "**all_base**",
+    -- Only dynamic fields are included as InfluxDB data points, while
+    -- the base fields serve as tags for them. If skip_fields does not
+    -- define any base fields, they are added to the fields of each data
+    -- point and each dynamic field value is set as the "value" field.
+    -- Adding "**all_base**" is recommended to avoid redundant data being
+    -- stored in each data point (the base fields as fields and as tags).
+    local points = {}
+    while true do
+        -- Iterate through Fields array in the message
+        local typ, name, value, representation, count = read_next_field()
+        -- Exit the perpetual loop when the iteration is complete
+        if not typ then break end
+
+        -- Only process fields that are not requested to be skipped
+        if not skip_fields_str or not skip_fields[name] then
+            -- Set the name attribute of this table by concatenating name_prefix
+            -- with the name of this particular field
+            local field_name = name_prefix.."."..name
+
+            -- Merge existing base fields with field in this iteration
+            local fields = {}
+            for k,v in pairs(columns) do fields[v] = values[k] end
+
+            -- Structure the table to match the expected Influxdb structure
+            fields["value"] = value
+            points[points_index] = {
+                name = field_name,
+                fields = fields
+            }
+
+            -- Increment the index to the table so the next iteration will
+            -- append another entry to the array. This allows us to send a
+            -- bunch of metrics to InfluxDB with a single HTTP request
+            points_index = points_index + 1
+        end
+
+        -- Include the dynamic fields as tags if they are defined in
+        -- configuration or the magic value "**all**" is defined.
+        -- Convert value to a string as this is required by the API
+        if used_tag_fields[name] or used_tag_fields["**all**"] then
```

Comment:
Here's another place where a boolean could be set at config loading time, and checked before the individual field lookup to avoid unnecessary hash checks.